Public/New-DirectoryBrowser.ps1

```diff
@@ -0,0 +1,74 @@
+function New-DirectoryBrowser {
+
+    <#
+    .SYNOPSIS
+        Renders a directory browser in HTML
+    .DESCRIPTION
+        Creates HTML that can be used as a directory browser
+    .PARAMETER FileSystemPath
+        The file path of the directory you would like to generate HTML
+    .PARAMETER HeaderName
+        The name you would like displayed at the top of the directory browser
+    .PARAMETER DirectoryBrowserPath
+        The current path in the directory browser relative to the root of the directory
+        browser (not the root of the site).
+    .PARAMETER FileSystemRootFolder
+        The root of the directory browser on the file system
+    .PARAMETER WebServerPath
+        The current path of the directory browser relative to the root of Polaris
+#>
+
+    param (
+
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Directory Path')]
+        [string]$FileSystemPath,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Header Name')]
+        [string]$HeaderName,
+
+        [Parameter(
+            Mandatory = $false,
+            HelpMessage = 'Directory Browser Path')]
+        [string]$DirectoryBrowserPath,
+
+        [string]$FileSystemRootFolder,
```

Comment:
I'll give it a shot. Here's an example that hopefully clarifies things a bit:

I register a static route at "/MyStaticFiles" serving files from C:\MyFolder which has the following files hosted at localhost:3000:
```
bootstrap/
├── css/
│   ├── bootstrap.css
│   ├── bootstrap.min.css
│   ├── bootstrap-theme.css
│   └── bootstrap-theme.min.css
├── js/
│   ├── bootstrap.js
│   └── bootstrap.min.js
└── fonts/
    ├── glyphicons-halflings-regular.eot
    ├── glyphicons-halflings-regular.svg
    ├── glyphicons-halflings-regular.ttf
    └── glyphicons-halflings-regular.woff 
```

1. A request comes in for http://localhost:3000/MyStaticFiles

* FileSystemPath = C:\MyFolder
* DirectoryBrowserPath = /
* FileSystemRootFolder = C:\MyFolder
* WebServerPath = /MyStaticFiles

2. A request comes in for http://localhost:3000/MyStaticFiles/bootstrap/css

* FileSystemPath = C:\MyFolder\bootstrap\css
* DirectoryBrowserPath = /bootstrap/css
* FileSystemRootFolder = C:\MyFolder
* WebServerPath = /MyStaticFiles/bootstrap/css

Does that clarify?

A couple of other notes on this that may explain why it's an odd one:

1. The function basically only returns HTML text given some inputs. I would much rather have made this a private internal function as it doesn't really do anything but since the scriptblocks don't execute in the module scope you have to make it a public function to have access to it in the script blocks.
2. I think most of the complexity around it comes from the theory that you could have multiple static routes serving from multiple "roots" tied to multiple folders on the hard drive. So basically, if I have the ```FileSystemPath``` how do I know when to stop offering a .. link in the browser? I could just keep offering a .. but at some point it may drop them out of the directory browser and back into another Polaris route.

Reply:
I think it's all better now. check it out.